.github/workflows/generate-changelog.yml

```diff
@@ -0,0 +1,30 @@
+name: Generate CHANGELOG
+on:
```

Comment:
@oarbusi cc @AgustinBettati as this is creating a commit in master there might be a race condition, e.g. if 2 PRs are merged almost at the same time. 

We might want to use concurrency limit for all GH actions writing to master: https://docs.github.com/en/actions/using-jobs/using-concurrency

e.g.:

concurrency:
  group: master-write
  cancel-in-progress: false

Reply:
(for example release process might also use this concurrency group)

Reply:
I might be wrong but in the case you explained:

> - job 2 starts, get 1 and 2, and it's faster than job 1 and commits
here changes from PR 1 and 2 are commited (and pushed)

> - job 1 commits, info about PR 2 is lost
job 1 would fail to push because tip of current branch would be behind its remote counterpart

so I think the issue would not happen in this specific case.
If the order was reversed (job 1 commits and pushes before job 2) the push from job 2 would fail and PR2 info would be "lost". This would be easily fixable by manually triggering the workflow.
All in all I think it a very unlikely edge case but we can use concurrency as you suggested

Reply:
you brought an interesting point, how can we know if this GH fails?  if it runs and fail i don't think we'll be watching (manually) if they're working

Reply:
To make sure I understand, the concurrency group configuration would prevent this potential push failure as one wound run after the other right?
It is true we will likely not keep an eye on the execution of this workflow, would a slack message to our notifications channel be sufficient? Not suggesting to do as part of this PR of course.

Reply:
Good point, we won't be actively looking at this GH action failing.
I see two ways of avoiding forgetting about this GH action failing
- Add a step to notify team/oncall person when it fails
- Trigger this workflow as part of the release process/workflow. 
I fell the second option is better because we avoid disrupting team/oncall with a notification where the fix is trivial(retrigger workflow) and we automate the fix. If in the release process the step fails we would notice it because when releasing whoever is working on it will notice and will fix it.
WDYT?

Reply:
I like the idea of keeping the changelog up-to-date after each PR as it will update the `Unreleased` section of changelog (helpful for us and external users).
I would not anticipate this workflow having frequent failures, meaning it would no be disruptive for on call. Seeing [cloudflares history](https://github.com/cloudflare/terraform-provider-cloudflare/actions/workflows/generate-changelog.yml) for this action the only failures I see is because of `Updates were rejected because the remote contains work that you do not have locally` (exact issue we where discussing here).

Reply:
@AgustinBettati yes, concurrency is like a mutex so only one job is executed and the others wait (similar to the mutex in ProjectIDExecution).   

i don't dislike the idea of notifying in Slack when this fails as i don't anticipate many errors, and it can help us to understand better the problems

Reply:
about "Trigger this workflow as part of the release process/workflow" we could do that in addition to do it in each PR in case the latest ones failed, but it's not probably worth it and it's better just to get notified if it fails

Reply:
Thanks for the input! Seems like we reached a decision: Notifying through slack when the job fails. I can do it in a follow up PR. Thanks again

Reply:
ok not to implement concurrency at the moment and see if we get errors